### ISSUE TITLE
Named tunnels wont accept no-tls-verify

### ISSUE THREAD

**Issue by seamooose** [cloudflare/cloudflared#277]
I am using named tunnels (https://blog.cloudflare.com/many-services-one-cloudflared/) for my configuration, and have found that no matter what I do, it appears impossible for the ingress rules to take the "no-tls-verify" option.

An snip of my config below (redacted of course):

```
tunnel: my_tunnel
credentials-file: /etc/cloudflared/<ID>.json
ingress:
# host1
 - hostname: host1.domain.com
   service: https://192.168.100.254
   no-tls-verify: yes
 # Catch-all rule doesn’t actually use any of the config
 - service: http_status:404
```

Despite this, and no matter where I put "no-tls-verify" in my config, I continue to receive:
`Error proxying request to origin: x509: certificate signed by unknown authority`

Is this intentional or does the no-tls-verify option belong somewhere else?

**Comment by seamooose** [comment]
typo

**Comment by TownLake** [comment]
Thanks for the question. You'll want to [specify it as a boolean](https://developers.cloudflare.com/argo-tunnel/configuration/config) for arguments that do not expect a value.

In this example
```
no-tls-verify: true
```

Let us know if that solves it for you.

**Comment by KaraRyougi** [comment]
I have the same issue. 
`Proxying to ingress 0 error: Error proxying request to origin: x509: cannot validate certificate for x.x.x.x because it doesn't contain any IP SANs`

I tried `no-tls-verify: true` and, according to [this](https://developers.cloudflare.com/argo-tunnel/configuration/ingress) document, `noTLSVerify: true`. Both options do not work.

**Comment by seamooose** [comment]
> Thanks for the question. You'll want to [specify it as a boolean](https://developers.cloudflare.com/argo-tunnel/configuration/config) for arguments that do not expect a value.
> 
> In this example
> 
> ```
> no-tls-verify: true
> ```
> 
> Let us know if that solves it for you.

Sadly not, tried that as well :-(

**Comment by vidarak** [comment]
Tried this? 
`tunnel: my_tunnel
credentials-file: /etc/cloudflared/<ID>.json
ingress:
# host1
 - hostname: host1.domain.com
   service: https://192.168.100.254
   originRequest:
     noTLSVerify: true
 # Catch-all rule doesn’t actually use any of the config
 - service: http_status:404`

**Comment by KaraRyougi** [comment]
Thank you! It works!

**Comment by seamooose** [comment]
+1 it works!

Thanks @vidarak

**Comment by adamchalmers** [comment]
Sorry, the CLI args are slightly different in some cases from the way you specify them under the `originRequest` key. See the list here: https://developers.cloudflare.com/argo-tunnel/configuration/ingress#noTLSVerify

**Comment by adamchalmers** [comment]
Here's a proposal: what if, instead of silently accepting the unrecognized key (`no-tls-verify`), instead, cloudflared logged an error and quit, telling you exactly which line was invalid, and linking you to the [docs for configuring ingress rules](https://developers.cloudflare.com/argo-tunnel/configuration/ingress#noTLSVerify)?

**Comment by vidarak** [comment]
That's good, but why not accept no-tls-verify like before on the global scope of the config (applicable to all origins specified for ingress... if multiple)? That would make it backwards compatible.

**Comment by adamchalmers** [comment]
The yaml library we're using in Go doesn't let us specify multiple aliases for keys, so we had to make a choice of which case to accept. names-like-this seem idiomatic for CLIs, but I thought for YAML, namesLikeThis are more common.

I figured that this isn't a true backwards-incompatibility problem, as it's an all-new block of YAML (under ingress.originRequest, not at the root of the document). So old cloudflareds would still keep working. I understand it's annoying to have to learn the new aliases! It's a tough tradeoff and we might have made the wrong one. But hopefully going forwards, anyone affected by this will get a nice error message linking them to an explanation.
